### ISSUE TITLE
Wrong query at Catalog

### ISSUE THREAD

**Issue by alasarr** [CartoDB/cartoframes#1084]
Query:
>Catalog().country('usa').category('demographics').geography('carto-do-public-data.tiger.geography_usa_countyclipped_2015').datasets

Query:

```sql
SELECT view.* FROM datasets_public view 
WHERE 
view.id IN ('carto-do.ags.demographics_sociodemographic_usa_q17_v1_yearly_2019','carto-do.carto.geography_usa_q17_v1') 
AND view.country_id = 'usa' AND view.category_id = 'demographics' 
AND view.geography_id = 'carto-do-public-data.tiger.geography_usa_countyclipped_2015'

```

That's really odd, because sometimes the same code works. I think it could be some kind of race condition
[catalog_alberto_issue.html.zip](https://github.com/CartoDB/cartoframes/files/3709567/catalog_alberto_issue.html.zip)

**Comment by alasarr** [comment]
Just found another one:
>catalog.country('usa').category('demographics').datasets.to_dataframe()
```sql
SELECT view.* FROM datasets_public view WHERE view.id IN ('carto-do.ags.demographics_sociodemographic_usa_q17_v1_yearly_2019','carto-do.carto.geography_usa_q17_v1') AND view.country_id = 'usa' AND view.category_id = 'demographics' AND view.geography_id = 'carto-do-public-data.tiger.geography_usa_countyclipped_2015'
```

**Comment by esloho** [comment]
This behavior is related to the new logic for purchases (now subscriptions) and it's been addressed in #1079

**Comment by alasarr** [comment]
@esloho I'm still able to see this issue at Google Colab some times
```
catalog.country('usa').category('demographics').geography('ags_blockgroup_1c63771c').datasets
```

```
SELECT t.* FROM geographies_public t WHERE t.slug = 'ags_blockgroup_1c63771c'
SELECT t.* FROM datasets_public t WHERE t.id IN ('carto-do.ags.demographics_sociodemographic_usa_q17_v1_yearly_2019','carto-do.carto.geography_usa_q17_v1') AND t.country_id = 'usa' AND t.category_id = 'demographics' AND t.geography_id = 'carto-do.ags.geography_usa_blockgroup_2015'
```

**Comment by esloho** [comment]
It''ll be corrected once #1079 is finished, I think that will be soon :)

**Comment by Jesus89** [comment]
#1079 is merged
